index.js

```diff
@@ -1,88 +1,130 @@
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
-
-// console.log(name)
+console.log("%cQuestion 2", "color: red") 
+const name = "Diana"
+console.log(name)
 // => "Raffy"
 // name = "Not Raffy" 
 // => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red") 
+
+const drinkWater = currentThirstLevel => {
+  console.log("Man I sure am thirsty")
+  currentThirstLevel --
+  console.log("Ahh that hits the spot")
+  return currentThirstLevel
+}
 
-// console.log(drinkWater(12))
-// // => "Man I sure am thirsty"
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
+console.log(drinkWater(12))
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red") 
+
+
+const sameSameButDifferent = (num, maybeNum) => {
+  if (num === maybeNum) {return "same same"}
+  else if (num == maybeNum) {return "same same (but different)"}
+  else {return "different"}
+}
 
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
+// * The solution below more literally fits the directions  *
+// const sameSameButDifferent = (num, maybeNum) => {
+//   if ((typeof(num) && typeof(maybeNum)) == "number" && num === maybeNum){
+//     return "same same"
+//   }
+//   else if ((typeof(num) != typeof(maybeNum)) && parseFloat(num) === parseFloat(maybeNum)) {
+//     return "same same (but different)"
+//   }
+//   else{
+//     return "different"
+//   }
+// }
+
+console.log(sameSameButDifferent(5, 5)) 
+// => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
-// // => "same same (but different)"
+console.log(sameSameButDifferent(123, "123")) 
+// => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
-// // => "different"
+console.log(sameSameButDifferent(5, 7)) 
+// => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
-// // => "different"
-// console.log("%c----------", "color: red") 
+console.log(sameSameButDifferent(123, "122")) 
+// => "different"
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
-// // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+console.log("%cQuestion 5", "color: red")
+const student1 = { name: "Duane", grade: 88 }
+
+const updateGrade = (student, grade) => {
+  student.grade = grade
+  return student
+}
+
```

Comment:
Good :+1: The updates made to the `student` object inside the function is reflected in the `student1` object because objects are ["call by sharing" unlike primitive data types](https://stackoverflow.com/questions/518000/is-javascript-a-pass-by-reference-or-pass-by-value-language).